pkg/runtime/runtime.go

```diff
@@ -460,9 +584,7 @@ func (a *DaprRuntime) initRuntime(ctx context.Context) error {
 		log.Infof("app max concurrency set to %v", a.runtimeConfig.appConnectionConfig.MaxConcurrency)
 	}
 
-	a.appHealthReady = func() {
-		a.appHealthReadyInit(ctx)
-	}
+	a.appHealthReady = a.appHealthReadyInit
```

Comment:
This was a lambda on purpose, otherwise `a` isn't updated. You can add the context but it needs to remain a lambda

Reply:
Not sure I follow- this is still a lambda, it just now has the same signature so doesn't need to be wrapped.

Reply:
I am not 100% sure of why that was even happening, but the original code had something like `a.appHealthReady = a.appHealthReadyInit` and it "captured" `a`, so if a property of `a` changed, it was not reflected in the method.